### ISSUE TITLE
Inconsistent shadow behaviour for some documents and images

### ISSUE THREAD

**Issue by janxkoci** [elementary/files#1111]
## Prerequisites
- [ ] I have searched open and closed issues for duplicates.

## Describe the bug
Some document and image formats exhibit inconsistent behaviour in regard to dropping shadows in Files view. The odd formats include: DJVU, EPS, ODG.

The particular issues I found:

- Djvu documents drop shadows with generic icon, with checkerboard background underneath.
- EPS images/documents(?) drop shadows, but they don't render image, just show a generic icon. Checkerboard background depends on transparency (I think).
- ODG image (LibreOffice Draw project) renders image but doesn't drop shadow.
- XCF image (GIMP project) drops shadow with checkerboard background, but shows icon instead of rendering image.

This is regardless of the view - columns, list, and icon views all have this problem. PDF or postscript (PS) documents don't do this. Note that encapsulated PS, aka EPS, do drop shadows, but they are mostly used as image format, not document format, particularly for vector graphics, so I guess the shadow is fine. But it would be better if it rendered the image instead of showing generic icon.

### To Reproduce
1. Navigate to folder with any DJVU document or EPS image in Files.
2. There!

### Expected behavior

- DJVU documents don't drop shadows nor checkerboard background.
- EPS images should render images instead of generic icon, or shouldn't drop shadow with checkerboard background.
- ODG should drop shadow.
- XCF should either render image or not drop shadow & checkerboard.

### Screenshots or screen recordings

![Snímek z 2019-11-22 12-22-09](https://user-images.githubusercontent.com/19331850/69421972-c26a0a80-0d22-11ea-9c7d-27279be935a1.png)

## Platform Information

![Snímek z 2019-11-22 12-25-58](https://user-images.githubusercontent.com/19331850/69422162-4de39b80-0d23-11ea-9083-f8d407b27828.png)

- [ ] I'm using the latest version from git that I've manually compiled
- [ ] I'm using the latest released stable version

## Additional context

Note that these formats by default open in Evince as documents: PDF, DJVU, PS, EPS. First three are considered document formats while EPS is mostly used for vector graphics. PDF and PS are also sometimes used for graphics. I have never seen DJVU file be anything else than ebook though.

Related bug reports: https://github.com/elementary/files/issues/570 and https://github.com/elementary/switchboard-plug-pantheon-shell/issues/131

**Comment by jeremypw** [comment]
Yes, there's something odd going on. There should only be checkerboard background for thumbnails with transparent edges, not for icons.  Presumably an issue with the IconRenderer.  As far as drop shadows are concerned these are incorporated in the icons and are not added by Files.  What is shown depends on 1) whether there is a thumbnailer backend for that file type 2) Whether there is a specific icon for that mime-type. 3) Which generic icon Files decides to use.

**Comment by jeremypw** [comment]
Confirmed still the case on OS8 at least for `.djvu` and `.eps` files.  These properties of these file show the mime type as `image/....` but they are not thumbnailed. Both have the checkerboard background that should only appear for thumbnails not icons.

**Comment by jeremypw** [comment]
I am inclined to leave this until the Gtk4 rewrite as the affected file types are not that common.
